### PR TITLE
PIM-5112: Export only the products updated since the last export

### PR DESCRIPTION
**Pull request by skeleton**
| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | :white_check_mark: 
| Added Behats                      | :white_check_mark: 
| Changelog updated                 | :white_check_mark: 
| Micro Demo to the PO (Story only) | :white_check_mark: 
| Migration script                  | NO
| Tech Doc                          | NO

### CODE REVIEW

File: features/export/product-export-builder/export_products_by_specific_date.feature [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    And I wait for the "csv_footwear_product_export" job to finish
+    Then exported file of "csv_footwear_product_export" should contain:
+      """
+      sku;categories;color;description-en_US-mobile;enabled;family;groups;lace_color;manufacturer;name-en_US;price-EUR;price-USD;rating;side_view;size;top_view;weather_conditions
+      SNKRS-1B;summer_collection;black;;1;sneakers;;;;"Model 1";50.00;70.00;;;45;;
+      SNKRS-1R;summer_collection;red;;1;sneakers;;;;"Model 1";50.00;70.00;;;45;;
+      """
+    When I edit the "SNKRS-1B" product
+    And I change the "Weather conditions" to "Hot"
+    Then I save the product
```
Comment (phaseinducer) [inline features/export/product-export-builder/export_products_by_specific_date.feature]: And

File: features/export/product-export-builder/export_products_by_specific_date.feature [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    When I am on the "csv_footwear_product_export" export job page
```
Comment (phaseinducer) [inline features/export/product-export-builder/export_products_by_specific_date.feature]: And

File: src/Akeneo/Component/Batch/Job/JobRepositoryInterface.php [hunk 3]
```diff
@@ -44,4 +44,13 @@ public function updateJobExecution(JobExecution $jobExecution);
...
     public function updateStepExecution(StepExecution $stepExecution);
+
+    /**
+     * Get the last complete job execution
+     *
+     * @param JobInstance $jobInstance
+     *
+     * @return JobExecution|null
+     */
+    public function getLastCompleteJobExecution(JobInstance $jobInstance);
```
Comment (phaseinducer) [inline src/Akeneo/Component/Batch/Job/JobRepositoryInterface.php]: BC break

File: src/Pim/Component/Connector/Reader/ProductReader.php [hunk 4]
```diff
@@ -58,12 +59,19 @@ class ProductReader extends AbstractConfigurableStepElement implements ItemReade
     /** @var string */
     protected $enabled;
 
+    /** @var string */
+    protected $updated;
+
+    /** @var JobRepositoryInterface */
+    protected $jobRepository;
```
Comment (phaseinducer) [inline src/Pim/Component/Connector/Reader/ProductReader.php]: Move it after `$objectDetacher` please

File: src/Pim/Component/Connector/Reader/ProductReader.php [hunk 5]
```diff
@@ -281,4 +335,26 @@ protected function rawToStandardProductStatus($rawStatus)
 
         return $status;
     }
+
+    /**
+     * Convert the UI product updated to the standard product updated
+     *
+     * @param string $rawUpdated
+     *
+     * @return DateTime|null
```
Comment (phaseinducer) [inline src/Pim/Component/Connector/Reader/ProductReader.php]: `\DateTime`

**Comment by phaseinducer** [comment]
Little comments & GTM

File: src/Akeneo/Bundle/BatchBundle/Job/DoctrineJobRepository.php [hunk 6]
```diff
@@ -127,4 +128,22 @@ public function updateStepExecution(StepExecution $stepExecution)
         $this->jobManager->persist($stepExecution);
         $this->jobManager->flush($stepExecution);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getLastCompleteJobExecution(JobInstance $jobInstance)
```
Comment (nidup) [inline src/Akeneo/Bundle/BatchBundle/Job/DoctrineJobRepository.php]: We could take inspiration from this http://docs.spring.io/spring-batch/apidocs/org/springframework/batch/core/repository/JobRepository.html

We still have several difference in our implem which limit us (we can't copy / paste this logic), I'm available to discuss about this IRL.

**Comment by nidup** [comment]
Really nice to see that we're able to implement such nice feature with such few code, API FTW \o/
I merged https://github.com/akeneo/pim-community-dev/pull/4472 I'll be available to rebase with you if you want because you'll have a small conflict on the configuration part.

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/DateFilter.php [hunk 7]
```diff
@@ -164,6 +164,7 @@ protected function formatSingleValue($type, $value)
         }
 
         if ($value instanceof \DateTime) {
+            $value->setTimezone(new \DateTimeZone('UTC'));
```
Comment (SamirBoulil) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/DateFilter.php]: I'm just wondering why we set the timezone ? Don't we use the php timezone configuration by default ?

Comment (skeleton) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/DateFilter.php]: Yes we use the PHP timezone configured in server to display dates in the application. but dates are always store with UTC timezone in database. that's why when we query it, we force timezone

**Pull request merged**